src/main/java/org/folio/service/manager/inputdatamanager/InputDataManagerImpl.java

```diff
@@ -82,24 +88,63 @@ protected void initBlocking(JsonObject request, JsonObject params) {
     FileDefinition fileExportDefinition = createExportFileDefinition(requestFileDefinition);
+    String jobExecutionId = requestFileDefinition.getJobExecutionId();
     String tenantId = okapiConnectionParams.getTenantId();
+    Optional<JobExecution> optionalJobExecution = jobExecutionService.getById(jobExecutionId, tenantId).result();
     Iterator<List<String>> readFileContent = sourceReader.getFileContentIterator(requestFileDefinition, getBatchSize());
     if (readFileContent.hasNext()) {
-      fileDefinitionService.save(fileExportDefinition, tenantId)
-        .compose(savedFileExportDefinition -> {
-          //get executionJob and update exportresult to in-progress
-          String jobExecutionId = requestFileDefinition.getJobExecutionId();
+      fileDefinitionService.save(fileExportDefinition, tenantId).compose(savedFileExportDefinition -> {
+        if (optionalJobExecution.isPresent()) {
           initInputDataContext(readFileContent, jobExecutionId);
           ExportPayload exportPayload = createExportPayload(okapiConnectionParams, savedFileExportDefinition, jobExecutionId);
           exportNextChunk(exportPayload, readFileContent);
+          updateJobExecutionStatusAndExportedFiles(optionalJobExecution.get(), savedFileExportDefinition, tenantId);
           return Future.succeededFuture();
-        });
+        } else {
+          return Future.failedFuture(format("Unable to update the job execution. JobExecution with id: '%s' doesn't exist", jobExecutionId));
+        }
+      });
     } else {
       fileDefinitionService.save(fileExportDefinition.withStatus(FileDefinition.Status.ERROR), tenantId);
-      //update job execution with error exportresult
+      updateJobStatusAndCompletedDate(optionalJobExecution, JobExecution.Status.FAIL, tenantId);
     }
   }
 
+  /**
+   * Update job execution status with FAIL && update completed date
+   *
+   * @param optionalJobExecution job execution
+   * @param status               status to update
+   * @param tenantId             completed date to update
+   */
+  private void updateJobStatusAndCompletedDate(Optional<JobExecution> optionalJobExecution, JobExecution.Status status, final String tenantId) {
+    if (optionalJobExecution.isPresent()) {
+      JobExecution jobExecution = optionalJobExecution.get();
+      jobExecution.setStatus(status);
+      jobExecution.setCompletedDate(new Date());
+      jobExecutionService.update(jobExecution, tenantId);
+    }
+  }
+
+  /**
+   * Update jobExecution status with IN-PROGRESS && update exported files && update completed date
+   *
+   * @param jobExecution         job execution
+   * @param fileExportDefinition definition of the file to export
+   * @param tenantId             tenant id
+   */
+  private void updateJobExecutionStatusAndExportedFiles(JobExecution jobExecution, FileDefinition fileExportDefinition, String tenantId) {
+    ExportedFile exportedFile = new ExportedFile()
+      .withFileId(fileExportDefinition.getId())
```

Comment:
we shouldn't be re-using the UUIDs, let;s have a unique Id for fileId

Reply:
Thank you for note, i will fix it